test/evp_extra_test.c

```diff
@@ -1623,6 +1623,47 @@ static int test_keygen_with_empty_template(int n)
     return ret;
 }
 
+/*
+ * Test that we fail if we attempt to use an algorithm that is not available
+ * in the current library context (unless we are using an algorithm that should
+ * be made available via legacy codepaths).
```

Comment:
So essentially, we're looking for legacy implementations that *should* have a provider implementation.  Would you mind writing the comment in those terms?  I find the current comment harder to grasp.